src/rax.h

```diff
@@ -183,6 +185,11 @@ typedef struct raxIterator {
     raxNode *node;          /* Current node. Only for unsafe iteration. */
     raxStack stack;         /* Stack used for unsafe iteration. */
     raxNodeCallback node_cb; /* Optional node callback. Normally set to NULL. */
+    size_t child_offset;    /* current child offset */
+    size_t cpos_max;        /* max number of child offsets the current stack can hold */
+    size_t cpos;            /* current position in the stack of child offsets */
+    uint8_t *child_offset_stack; /* stack (vector) of child offsets */
+    uint8_t child_offset_stack_static[RAX_ITER_CHILD_STATIC_LEN];
```

Comment:
Shouldn't they be moved to `raxStack`? It looks like `child_offset_stack` and `raxStack->stack` must be the same length.

Reply:
I believe this may be the case. I'll have time to review this and other suggestions this week.

Reply:
I consolidated the offset vector into `raxStack` and synced it with `raxStack->stack`. `raxIterator->child_offset` remains in place as it and `raxIterator->node` represent the current state of the iterator which is pushed to or popped from the stack. Other changes were done to support the new approach. The tests in redis/redis pass as well as the fuzz tests in antirez/rax.